### PR TITLE
feat: Pokemon graphQL section

### PR DESCRIPTION
**Pull request by mrestuccia**

### CODE REVIEW

File: app/graphql/fragments/pokemon-full.graphql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+fragment fullPokemon on Pokemon {
+  pokemonNumber
```
Comment (cml391) [inline app/graphql/fragments/pokemon-full.graphql]: are pokemonNumber and id the same? If so, you should make the naming consistent

File: server/graphql/make-executable-schema.ts [hunk 2]
```diff
@@ -4,9 +4,11 @@ import path from 'path';
 import 'regenerator-runtime/runtime';
 import config from '../config';
 
+import { pokemon, pokemonAll, pokemonCreate, pokemonDelete, pokemonEdit } from './pokemon-resolver';
+
 export const resolveFunctions = {
-  RootQueryType: {},
-  RootMutationType: {},
+  RootQueryType: { pokemonAll, pokemon },
```
Comment (cml391) [inline server/graphql/make-executable-schema.ts]: small thing, but with graphql queries we tend to do pluralization to do `pokemonAll`, so having `pokemon` and `pokemons`

File: server/graphql/pokemon-resolver.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  input: IPokemonCreateInput;
+}
+
+export async function pokemonCreate(
+  source: any,
+  { input }: IPokemonCreateOnRootMutationTypeArguments,
+  { testTransaction }: IContext,
+): Promise<IRootMutationType['pokemonCreate']> {
+  return transaction(testTransaction || Pokemon.knex(), async txn => {
+    const createdPokemon = await Pokemon.create(input, txn);
```
Comment (cml391) [inline server/graphql/pokemon-resolver.ts]: another small thing, but you can just `return Pokemon.create(input, txn);`

File: server/models/Pokemon.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+      deletedAt: { type: 'timestamp' },
+    },
+  };
+
+
+  // Relations
+  static relationMappings: RelationMappings = {
+    item: {
+      relation: Model.HasManyRelation,
+      modelClass: join(__dirname, 'Item'),
```
Comment (cml391) [inline server/models/Pokemon.ts]: for this, you can just do `modelClass: Item` which should simplify this a little?

File: app/graphql/fragments/pokemon-full.graphql (hunk 1 above)
Comment (cml391) [reply app/graphql/fragments/pokemon-full.graphql]: whoops, I see later on that they are different, all good

File: server/graphql/pokemon-resolver.ts (hunk 3 above)
Comment (mrestuccia) [reply server/graphql/pokemon-resolver.ts]: If I do return directly. Will that enforce the await?

Comment (mrestuccia) [reply server/graphql/pokemon-resolver.ts]: I just check. It works. Making the change.

File: server/models/Pokemon.ts (hunk 4 above)
Comment (mrestuccia) [reply server/models/Pokemon.ts]: Done.

File: server/graphql/make-executable-schema.ts (hunk 2 above)
Comment (mrestuccia) [reply server/graphql/make-executable-schema.ts]: done.

**Pull request merged**